src/contexts/ListsContext.tsx

```diff
@@ -69,24 +83,35 @@ export const ListsProvider = ({ children }: { children: React.ReactNode }) => {
     );
   };
 
-  const deleteTask = (listId: number, taskId: number) => {
+  const deleteTask = (listId: number | string, taskId: number | string) => {
     setLists((prevLists) =>
       prevLists.map((list) =>
         list.id === listId
-          ? { ...list, tasks: list.tasks.filter((task) => task.id !== taskId) }
+          ? {
+              ...list,
+              taskCount: list.taskCount - 1,
+              tasks: list.tasks.filter((task) => task.id !== taskId),
+            }
           : list,
       ),
     );
   };
 
+  const getListByListId = (listId: number | string) => {
+    return lists.find((item) => item.id === listId) ?? undefined;
```

Comment:
Why do we need `?? undefined`, if it cannot find the list it will automatically return undefined.